lib/project-data.ts

```diff
@@ -42,45 +40,48 @@ export class ProjectData implements IProjectData {
 
 	constructor(private $fs: IFileSystem,
 		private $errors: IErrors,
-		private $logger: ILogger,
 		private $projectHelper: IProjectHelper,
 		private $staticConfig: IStaticConfig,
 		private $options: IOptions) { }
 
-	public initializeProjectData(projectDir? :string): void {
+	public initializeProjectData(projectDir?: string): void {
 		projectDir = projectDir || this.$projectHelper.projectDir;
 		// If no project found, projectDir should be null
 		if (projectDir) {
-			this.initializeProjectDataCore(projectDir);
+			const projectFilePath = path.join(projectDir, this.$staticConfig.PROJECT_FILE_NAME);
 			let data: any = null;
 
-			if (this.$fs.exists(this.projectFilePath)) {
+			if (projectFilePath) {
 				let fileContent: any = null;
 				try {
-					fileContent = this.$fs.readJson(this.projectFilePath);
+					fileContent = this.$fs.readJson(projectFilePath);
 					data = fileContent[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE];
 				} catch (err) {
-					this.$errors.fail({
-						formatStr: "The project file %s is corrupted." + EOL +
+					this.$errors.failWithoutHelp(
+						`The project file ${this.projectFilePath} is corrupted.` + EOL +
 						"Consider restoring an earlier version from your source control or backup." + EOL +
-						"Additional technical info: %s",
-						suppressCommandHelp: true
-					},
-						this.projectFilePath, err.toString());
+						`Additional technical info: ${err.toString()}`);
 				}
 
 				if (data) {
+					this.projectDir = projectDir;
+					this.projectName = this.$projectHelper.sanitizeName(path.basename(projectDir));
+					this.platformsDir = path.join(projectDir, constants.PLATFORMS_DIR_NAME);
```

Comment:
I like the idea, but there's a reason to set the properties here - we are accessing them in many, many places, so computing each of them every single time will require reading the directory, the package.json, etc. This will make a lot of I/O operations. I can cache the properties after first execution, but this will not work for Fusion, which is a long living process.
So I'll keep this code until I figure out a better way for handling this.